platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/Capture.kt

```diff
@@ -153,26 +157,28 @@ object Capture {
                 return
             }
 
-            // Ideally we would use `getAndUpdate` in here but it's available for API 24 and up only.
-            if (default.compareAndSet(LoggerState.NotStarted, LoggerState.Starting)) {
-                try {
-                    val logger =
-                        LoggerImpl(
-                            apiKey = apiKey,
-                            apiUrl = apiUrl,
-                            fieldProviders = fieldProviders,
-                            dateProvider = dateProvider ?: SystemDateProvider(),
-                            configuration = configuration,
-                            sessionStrategy = sessionStrategy,
-                            bridge = bridge,
-                        )
-                    default.set(LoggerState.Started(logger))
-                } catch (e: Throwable) {
-                    Log.w("capture", "Failed to start Capture", e)
-                    default.set(LoggerState.StartFailure)
+            sdkInitializationDispatcher.launchAsyncAndShutDown {
+                // Ideally we would use `getAndUpdate` in here but it's available for API 24 and up only.
+                if (default.compareAndSet(LoggerState.NotStarted, LoggerState.Starting)) {
```

Comment:
don't we run the risk to race conditions here (multiple loggers being initialized if the start() method is called in rapid succession)? I'd think that starting the background operation _after_ this check would avoid that